src/libsync/propagateupload.h

```diff
@@ -176,51 +261,66 @@ class PropagateUploadFile : public PropagateItemJob {
     int _currentChunk;
     int _chunkCount; /// Total number of chunks for this file
     int _transferId; /// transfer id (part of the url)
-    QElapsedTimer _duration;
-    QVector<AbstractNetworkJob*> _jobs; /// network jobs that are currently in transit
-    bool _finished; // Tells that all the jobs have been finished
 
-    // measure the performance of checksum calc and upload
-    Utility::StopWatch _stopWatch;
+    quint64 chunkSize() const { return _propagator->chunkSize(); }
 
-    QByteArray _transmissionChecksum;
-    QByteArray _transmissionChecksumType;
 
-    bool _deleteExisting;
+public:
+    PropagateUploadFileV1(OwncloudPropagator* propagator,const SyncFileItemPtr& item) :
+        PropagateUploadFileCommon(propagator,item) {}
 
-    quint64 chunkSize() const { return _propagator->chunkSize(); }
+    void doStartUpload() Q_DECL_OVERRIDE;
 
-public:
-    PropagateUploadFile(OwncloudPropagator* propagator,const SyncFileItemPtr& item)
-        : PropagateItemJob(propagator, item), _startChunk(0), _currentChunk(0), _chunkCount(0), _transferId(0), _finished(false), _deleteExisting(false) {}
-    void start() Q_DECL_OVERRIDE;
+private slots:
+    void startNextChunk();
+    void slotPutFinished();
+    void slotUploadProgress(qint64,qint64);
+};
 
-    bool isLikelyFinishedQuickly() Q_DECL_OVERRIDE { return _item->_size < 100*1024; }
+/**
+ * @ingroup libsync
+ *
+ * Propagation job, impementing the new chunking agorithm
+ *
+ */
+class PropagateUploadFileNG : public PropagateUploadFileCommon {
+    Q_OBJECT
+private:
+    quint64 _sent; /// amount of data (bytes) that was already sent
+    uint _transferId; /// transfer id (part of the url)
+    int _currentChunk; /// Id of the next chunk that will be sent
+    bool _removeJobError; /// If not null, there was an error removing the job
 
+    // Map chunk number with its size  from the PROPFIND on resume.
+    // (Only used from slotPropfindIterate/slotPropfindFinished because the LsColJob use signals to report data.)
+    QMap<int, quint64> _serverChunks;
+
+    quint64 chunkSize() const { return _propagator->chunkSize(); }
```

Comment:
Well, the idea would be that we could do dynamic chunking, so they would be different.